sklearn/preprocessing/data.py

```diff
@@ -138,8 +139,39 @@ def scale(X, axis=0, with_mean=True, with_std=True, copy=True):
         Xr = np.rollaxis(X, axis)
         if with_mean:
             Xr -= mean_
+            mean_1 = Xr.mean(axis=0)
+            # Verify that mean_1 is 'close to zero'. If X contains very
+            # large values, mean_1 can also be very large, due to a lack of
+            # precision of mean_. In this case, a pre-scaling of the
+            # concerned feature is efficient, for instance by its mean or
+            # maximum.
+            if isinstance(mean_1, np.ndarray):
+                if not np.allclose(mean_1, np.zeros(np.shape(mean_1)[0])):
```

Comment:
ok i'll change the np.allclose. The if is needed if we don't want to change X_r by useless subtraction (Xr -= mean_2 with a mean_2 very close to 0). But maybe the complexity of testing np.allclose is greater than just substracting, is that what you meant ?